### ISSUE TITLE
Cognito session token expire

### ISSUE THREAD

**Issue by SPradnya** [aws/aws-sdk-js#1685]
Hi,
Cognito session gets expired and it's not recognized till I manually check in browser network window. 
Is there any way to find my Cognito session is expired or not? I need to log out a user after token get's expired. Please suggest a solution.
Thanks in advance.

**Comment by AllanFly120** [comment]
Hi @SPradnya 
SDK does have a [Credentials.get()](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html#get-property) methods that check the existing credentials and refresh them when expired. If you are using other identity providers (like facebook/google etc.), you may need to manually load the updated tokens if `Credentials.get()` callback is called with an error. You may use SDKs from other identity providers (for AWS Cognito you can see [here](https://aws.amazon.com/documentation/cognito/)) to do so.

**Comment by SPradnya** [comment]
[Doc](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html#expireTime-property) says **Credentials.get()** helps to find that credentials have been expired. Will callback for this method will return an error for session expired?

**Comment by AllanFly120** [comment]
Basically, if you are using the cognito identity credential, the `get()` method will first check whether the present credential is expired by comparing the expire time and current time. If expired, it will automatically use [`cognitoidentity.getId`](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CognitoIdentity.html#getId-property). And if this fails, it will clear the expired Id the execute the call back with error in the last. 
So, to sum up, this callback is for handling the error in [`cognitoidentity.getId`](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CognitoIdentity.html#getId-property) and the expired credential is already taken care of. And the error is returned because the user's credential gets expired and SDK can not refresh it.

**Comment by srchase** [comment]
It looks like @AllanFly120's explanation covered the original question.
